recipes/swig/all/conanfile.py

```diff
@@ -1,170 +1,164 @@
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        # Rely on AutotoolsDeps instead of pcre2-config
+        # https://github.com/swig/swig/blob/v4.1.1/configure.ac#L70-L92
+        # https://github.com/swig/swig/blob/v4.0.2/configure.ac#L65-L86
+        replace_in_file(self, os.path.join(self.source_folder, "configure.ac"),
+                        'AS_IF([test "x$with_pcre" != xno],', 'AS_IF([false],')
 
     def build(self):
         self._patch_sources()
-        with tools.chdir(os.path.join(self._source_subfolder)):
-            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
-        with self._build_context():
-            autotools = self._configure_autotools()
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            self.run("./autogen.sh")
+            autotools.configure()
             autotools.make()
 
     def package(self):
-        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="COPYRIGHT", dst="licenses", src=self._source_subfolder)
-        self.copy("*", src="cmake", dst=self._module_subfolder)
-        with self._build_context():
-            autotools = self._configure_autotools()
+        copy(self, "LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "COPYRIGHT", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "*.cmake",
+             dst=os.path.join(self.package_folder, self._module_subfolder),
+             src=os.path.join(self.export_sources_folder, "cmake"))
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
             autotools.install()
-
-    @property
-    def _swiglibdir(self):
-        return os.path.join(self.package_folder, "bin", "swiglib").replace("\\", "/")
+        for path in self.package_path.iterdir():
+            if path.is_dir() and path.name not in ["bin", "lib", "licenses"]:
+                rmdir(self, path)
 
     @property
     def _module_subfolder(self):
         return os.path.join("lib", "cmake")
 
     @property
-    def _module_file(self):
-        return "conan-official-{}-targets.cmake".format(self.name)
+    def _cmake_module_rel_path(self):
+        return os.path.join(self._module_subfolder, "conan-swig-variables.cmake")
 
     def package_info(self):
-        self.cpp_info.includedirs=[]
+        self.cpp_info.includedirs = []
+        self.cpp_info.set_property("cmake_file_name", "SWIG")
+        self.cpp_info.set_property("cmake_target_name", "SWIG::SWIG")
+        self.cpp_info.set_property("cmake_build_modules", [self._cmake_module_rel_path])
+
+        self.buildenv_info.define_path("SWIG_LIB", os.path.join(self.package_folder, "bin", "swiglib"))
```

Comment:
Same here, don't we need to add the bindir to the PATH, so that the `find_program(SWIG_EXECUTABLE swig)`  (from `recipes/swig/all/cmake/conan-swig-variables.cmake`) actually works?

Reply:
It's not necessary. That's only required for Conan v1 and `self.env_info`. `VirtualBuildEnv` adds bin dirs to PATH automatically.

Reply:
Does that only work as a tool_requires?

I tried with conan v2 today, defined it as a requires (not a tool_requires) and cmake definitely picked /usr/bin/swig and not the conan one, even though I used the conan_toolchain.cmake